presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpClientStats.java

```diff
@@ -0,0 +1,382 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server.remotetask;
+
+import com.facebook.airlift.stats.CounterStat;
+import com.facebook.airlift.stats.DistributionStat;
+import com.facebook.airlift.stats.TimeStat;
+import com.google.inject.Singleton;
+import org.weakref.jmx.Managed;
+import org.weakref.jmx.Nested;
+import reactor.netty.http.client.ContextAwareHttpClientMetricsRecorder;
+import reactor.util.context.ContextView;
+
+import java.net.SocketAddress;
+import java.time.Duration;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+@Singleton
+public class HttpClientStats
+        extends ContextAwareHttpClientMetricsRecorder
+{
+    // HTTP level metrics
+    private final TimeStat responseTime = new TimeStat();
+    private final TimeStat dataReceivedTime = new TimeStat();
+    private final TimeStat dataSentTime = new TimeStat();
+    private final CounterStat errorsCount = new CounterStat();
+    private final CounterStat bytesReceived = new CounterStat();
+    private final CounterStat bytesSent = new CounterStat();
+    private final DistributionStat payloadSize = new DistributionStat();
+
+    // Channel level metrics
+    private final TimeStat connectTime = new TimeStat();
+    private final TimeStat tlsHandshakeTime = new TimeStat();
+    private final TimeStat resolveAddressTime = new TimeStat();
+    private final CounterStat channelErrorsCount = new CounterStat();
+    private final CounterStat channelBytesReceived = new CounterStat();
```

Comment:
How frequently are `channelBytesReceived` / `channelBytesSent` updated? Is overhead significant?

Reply:
its for every byte sent ..so it get updated high number of times